tools/shared.py

```diff
@@ -542,7 +542,7 @@ def perform_sanify_checks():
       exit_with_error('The JavaScript shell used for compiling (%s) does not seem to work, check the paths in %s', COMPILER_ENGINE, EM_CONFIG)
 
   with ToolchainProfiler.profile_block('sanity LLVM'):
-    for cmd in [CLANG, LLVM_LINK, LLVM_AR, LLVM_OPT, LLVM_AS, LLVM_DIS, LLVM_NM, LLVM_INTERPRETER]:
```

Comment:
Should we still check for LLVM_LINK and LLVM_OPT in the fastcomp case? 

I guess as long as the error message when they are missing is OK, I'm fine with defering the error until point of need.

Reply:
Yeah, it seems simplest to defer the error, I agree. The early error here is more of "whether things look at all reasonable for anything to work". Later checks can be more precise (in particular, I'm not sure if the code here can tell if we are using fastcomp or not).